### PR TITLE
build: fix#2680 - update GoLang in all places to Go 1.15.2

### PR DESCRIPTION
**Pull request by jpwhitemn**
Updated Dockerfile.build, go.mod, README.md, snapcraft.yml to Go 1.15 or patch 1.15.2 where applicable.  Removed reference in Jenkinsfile to allow devops CI/CD default to apply

Signed-off-by: Jim White <jpwhite_mn@yahoo.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit subject follows the [Conventional Commits spec](https://github.com/zeke/semantic-pull-requests)
- [x] The commit message follows the [EdgeX Contributor Guide](https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] `make test` has completed successfully

also ran `make docker` and `snapcraft --debug`

## PR Type

What kind of change does this PR introduce?
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [x] Other... Please describe:  update GoLang version used for all of edgex-go

## What is the current behavior?
n/a

Issue Number: #2680  

## What is the new behavior?
n/a

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Are there any new imports or modules? If so, what are they used for and why?
none

## Are there any specific instructions or things that should be known prior to reviewing?
Removed go lang reference in Jenkinsbuild.  This will now default to what devops has defaulted in CI/CD.

### CODE REVIEW

**Comment by jpwhitemn** [comment]
recheck

**Comment by jpwhitemn** [comment]
recheck

File: Dockerfile.build [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 #
 
-ARG BASE=golang:1.13-alpine
+ARG BASE=golang:1.15.2-alpine
```
Comment (lenny-intel) [inline Dockerfile.build:17]: I don't thing we want to ping the patch. I verified that current `golang:1.15` has Go 1.15.2.

File: README.md [hunk 2]
```diff
@@ -51,8 +51,8 @@ on the snap, including how to install it, please refer to [EdgeX snap](https://g
 
 ##### Go
 
-- The current targeted version of the Go language runtime for release artifacts is v1.13.x
-- The minimum supported version of the Go language runtime is v1.13.x (currently v1.13)
+- The current targeted version of the Go language runtime for release artifacts is v1.15.x
+- The minimum supported version of the Go language runtime is v1.13.x
```
Comment (lenny-intel) [inline README.md:55]: This should be 1.15.x since we are forcing that in the go.mod files.

**Review by lenny-intel — CHANGES_REQUESTED** [review]
Need to update your commit and PR tile to replace `fix:` with `build` We also put a second line in the commit the indicates the issue #. Like `closes <issue number>`

File: README.md (hunk 2 above)
Comment (jpwhitemn) [reply README.md:55]: I think the readme we have always suggested min version and my thought was 1.13 was still ok to use.  I change it though.

Comment (lenny-intel) [reply README.md:55]: The version we put in the go.mod file specifies the min version, so in our case it will be 1.15.x

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by tonyespy — COMMENTED** [review]
This looks good to me, however I ran into a build failure dealing with `go.mod` and `vendor/modules.txt`. I think this is related to the existence of `/vendor` in my local sandbox, and getting rid of that seemed to fix the problem. I'm just doing one more sanity build, and if it works, I'll approve.

**Note** - go 1.14 introduced some extra code to sanity check go.mod dependencies.

**Comment by jpwhitemn** [comment]
recheck

**Review by tonyespy — DISMISSED** [review]
OK, deleting the local /vendor directory in my workspace did the trick.

All looks good to me, approving...

**Review by tonyespy — COMMENTED** [review]
Hmmm, seems like you're getting bitten by the same module sanity checking that was introduced in go 1.14.

That said,  I wonder if this is really the correct approach? If a module exists in `vendor/modules.txt` and is implicit, it still is getting statically linked into our code isn't it? If so, it needs attribution.

For the sake of expediting this PR, I'd be OK with opening an issue to track this.

One other comment is that your commit messages don't pass muster wrt to conventional commits.

> fix: fix#2680 - update GoLang in all places to Go 1.15.2 …

**suggest**:

```
fix(build): update to go 1.15

Updated Dockerfile.build, go.mod, README.md, snapcraft.yml to Go 1.15 or patch
1.15.2 where applicable.  Removed reference in Jenkinsfile to allow devops CI/CD
default to apply.

Fixes: #2680 
```

> build: fix #2680 - update everyhing to Go 1.15.2 …

**suggest**: fold into previous commit...
 
> modify test-attribution-txt.sh to ignore explicit …

**suggest**:

```
fix(build): modify attribution script to ignore implicit dependencies

go 1.14 added new module validation logic which breaks our attribution
shell script used to ensure we properly attribute module dependencies.
```

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by advancedwebdeveloper** [comment]
Please provide some support for [gollvm](https://go.googlesource.com/gollvm/) .

**Pull request merged**